src/dump.c

```diff
@@ -3047,6 +3059,20 @@ static int trace_method(jl_typemap_entry_t *entry, void *closure)
     return 1;
 }
 
+void jl_gc_mark_deserializer_state(jl_ptls_t ptls)
+{
+    size_t i;
+    for (i=0; i < flagref_list.len; ) {
+        jl_value_t **loc = (jl_value_t**)flagref_list.items[i];
+        if (loc)
+            jl_gc_push_root(ptls, *loc);
+        i += 2;
+    }
+    for (i=0; i < backref_list.len; i++) {
+        jl_gc_push_root(ptls, (jl_value_t*)backref_list.items[i]);
+    }
```

Comment:
we should never revisit the backref list after this point, so there's no need to root it